code/modules/reagents/reagent_containers.dm

```diff
@@ -67,7 +67,8 @@
 	return
 */
 /obj/item/weapon/reagent_containers/afterattack(obj/target, mob/user , flag)
-	return
+	if(user.a_intent == "harm")
+		return ..()

```

Comment:
I've done some tests.

With potato in hand, 
Clicking self  on help intent = Eat potato.
Clicking self on harm intent = Bash head in with potato.
Click other on help intent = Forcefeed potato.
Click other on harm intent = Bash head in with potato.